dovecot/tasks/main.yml

```diff
@@ -1,9 +1,21 @@
 ---
+- name: include OS-specific variables
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_os_family }}.yml"
+    - default.yml
+
 - name: install dovecot package
   package:
     name: dovecot
     state: installed
 
+- name: install dovecot-pigeonhole package
+  package:
+    name: dovecot-pigeonhole
+    state: installed
+  when: dovecot_sieve
+
```

Comment:
You could move this dovecot-pigeonhole task to the when: dovecot_sieve block below